vql/server/repository.go

```diff
@@ -95,6 +95,88 @@ func (self ArtifactSetFunction) Info(
 	}
 }
 
+type ArtifactDeleteFunctionArgs struct {
+	Name string `vfilter:"optional,field=name,doc=The Artifact to delete"`
+}
+
+type ArtifactDeleteFunction struct{}
+
+func (self *ArtifactDeleteFunction) Call(ctx context.Context,
+	scope *vfilter.Scope,
+	args *ordereddict.Dict) vfilter.Any {
+
+	arg := &ArtifactDeleteFunctionArgs{}
+	err := vfilter.ExtractArgs(scope, args, arg)
+	if err != nil {
+		scope.Log("artifact_delete: %v", err)
+		return vfilter.Null{}
+	}
+
+	config_obj, ok := artifacts.GetServerConfig(scope)
+	if !ok {
+		scope.Log("artifact_delete: Command can only run on the server")
+		return vfilter.Null{}
+	}
+
+	manager, _ := services.GetRepositoryManager()
+	if manager == nil {
+		scope.Log("artifact_delete: Command can only run on the server")
+		return vfilter.Null{}
+	}
+
+	global_repository, err := manager.GetGlobalRepository(config_obj)
+	if err != nil {
+		scope.Log("artifact_delete: %v", err)
+		return vfilter.Null{}
+	}
+
+	definition, pres := global_repository.Get(config_obj, arg.Name)
+	if !pres {
+		scope.Log("artifact_delete: Artifact %v not found", arg.Name)
+		return vfilter.Null{}
+	}
+
+	// Same criteria as
```

Comment:
I wonder if we actually need this restriction here at all. I think this function will be useful for artifact packs as well. As a rule the GUI is very conservative in what you can do because we don't expect a depth of knowledge about the system there, but vql does expose the sharp edges. For example artifact_set() allows you to set any artifact name including over riding an existing built in artifact (which is not a great idea generally so the GUI won't allow it).

Reply:
The only case i can think of is when a user is mature enough that their own content overshadows all the pre-built content, at which point they'll want to turn off the VQL shipped with vraptor. There is a similar thing is airflow, which allows you to set an config variable to not include the pre-included workflows: https://github.com/apache/airflow/blob/master/airflow/config_templates/default_airflow.cfg#L126

I know some of the VQLs are highly unrecommended to be modified (for example the generic.client.info one), so it may be worth splitting up the built-in VQL to core (like client.info) and then the other ones, which are example/built in content, that an advanced user can turn off with a config setting similar to airflow.